wikimedia/wikipedia-ios
# Update `themeAnalyticsName` values to match MobileWikiAppiOSUserHistory schema

**Phabricator:** https://phabricator.wikimedia.org/T289604

### Notes
* This PR updates the `themeAnalyticsName` method to send values that match our analytics schema, and to make sure it sends `dark` as a value when the Default theme is selected, but the device is set to dark mode. To ensure that, a validation is made with both the applied theme and the system appearance mode, to decide the correct value to be sent.

### Test Steps
1. Launch the app, select the Default theme
2. Set the device appearance to Light Mode, the value sent to analytics should be default.
3. Set the device appearance to Dark Mode, the value sent to analytics should be dark.
4. Set the theme to Light, the value sent should be Default
